lang/LangPrimSource/PyrFilePrim.cpp

```diff
@@ -1326,6 +1326,74 @@ int prPipeOpen(struct VMGlobals *g, int numArgsPushed)
 	return errNone;
 }
 
+int prPipeOpenArgv(struct VMGlobals *g, int numArgsPushed)
+{
+	PyrSlot *a, *b, *c;
+	char mode[12];
+
+	a = g->sp - 2;
+	b = g->sp - 1; //argv
+	c = g->sp;     //mode
+
+	if (NotObj(b)) return errWrongType;
+
+	if (NotObj(c) || !isKindOf(slotRawObject(c), class_string))
+		return errWrongType;
+	if (slotRawObject(c)->size > 11) return errFailed;
+
+	PyrFile *pfile = (PyrFile*)slotRawObject(a);
```

Comment:
could you use a `static_cast` here? that's more idiomatic for modern C++

Reply:
I tried, because cppcheck was flagging this, but it gives an error. I confess I don't understand well enough the difference between static_cast, dynamic_cast and C type cast to know why.

Reply:
```
error: invalid static_cast from type ‘PyrObject*’ to type ‘PyrFile*’
  PyrFile *pfile = static_cast<PyrFile*>(slotRawObject(a));
```

Reply:
ah ok, should be a reinterpret_cast then

Reply:
ok, changed.